research/conditional/moe_layers/expert_choice.py

```diff
@@ -50,6 +51,7 @@ def __init__(
         self.softmax_ungrouped = softmax_ungrouped
         self.n_gating_heatmaps = n_gating_heatmaps
         self.use_full_einsum = use_full_einsum
+        self.simulate_groups = group_size
```

Comment:
An assert would be cool that makes sure `full_einsum` and is `True` and we are grouping over position

Reply:
with this implementation, it will work without einsum